src/richie/apps/search/viewsets/categories.py

```diff
@@ -38,16 +38,19 @@ def list(self, request, version, kind):
 
         limit, offset, query = params_form.build_es_query(kind=kind)
 
-        # pylint: disable=unexpected-keyword-arg
-        query_response = ES_CLIENT.search(
-            _source=getattr(self._meta.indexer, "display_fields", "*"),
-            index=self._meta.indexer.index_name,
-            doc_type=self._meta.indexer.document_type,
-            body=query,
-            # Directly pass meta-params through as arguments to the ES client
-            from_=offset,
-            size=limit or getattr(settings, "RICHIE_ES_PAGE_SIZE", ES_PAGE_SIZE),
-        )
+        try:
+            # pylint: disable=unexpected-keyword-arg
+            query_response = ES_CLIENT.search(
+                _source=getattr(self._meta.indexer, "display_fields", "*"),
+                index=self._meta.indexer.index_name,
+                doc_type=self._meta.indexer.document_type,
+                body=query,
+                # Directly pass meta-params through as arguments to the ES client
+                from_=offset,
+                size=limit or getattr(settings, "RICHIE_ES_PAGE_SIZE", ES_PAGE_SIZE),
+            )
+        except NotFoundError:
+            return Response(status=404)
```

Comment:
did you try raising NotFound https://www.django-rest-framework.org/api-guide/exceptions/#notfound ?